src/js/libcs/Accessors.js

```diff
@@ -168,6 +168,22 @@ Accessor.setField = function(geo, field, value) {
         movepointscr(geo, List.turnIntoCSList([value.value[0], value.value[1], CSNumber.real(1)]), "homog");
     }
 
+    if (field === "x" && geo.kind === "P" && geo.movable && value.ctype === "number") {
+        if (CSNumber._helper.isAlmostZero(geo.homog.value[2])) {
+            console.log("cannot set x component without y component of far points");
+        } else {
+            movepointscr(geo, List.turnIntoCSList([value, CSNumber.div(geo.homog.value[1], geo.homog.value[2]), CSNumber.real(1)]), "homog");
+        }
+    }
```

Comment:
I've got an alternative suggestion: instead of `[value, y/z, 1]` you might use `[value*z, y, z]` as the new coordinate vector. That would allow you to drop the case distinction since you wouldn't have a division by zero and you would have well-defined (if not exactly geometrically meaningful) behavior even for points at infinity. And you would avoid the costly division, using the cheaper multiplication instead.